### ISSUE TITLE
Error 500s from remote exporter

### ISSUE THREAD

**Issue by louisvernon** [znerol/prometheus-pve-exporter#13]
## Summary of Issue
Running prometheus-pve-exporter on a non-proxmox server. prometheus-pve-exporter is returning 500s when queried.

## Environment
`PVE Manager Version pve-manager/5.2-1`
I am running the `prometheus-pve-exporter-1.1.1` pip release which I saw patched a similar issue.

## Detailed Description
prometheus-pve-exporter logs show:
```
[10/Jun/2018 19:28:41] "GET /pve?module=default&target=192.168.1.2 HTTP/1.1" 500 -
[10/Jun/2018 19:28:49] "GET /pve?module=default&target=192.168.1.4 HTTP/1.1" 500 -
[10/Jun/2018 19:28:50] "GET /pve?module=default&target=192.168.1.3 HTTP/1.1" 500 -
```
proxmox node syslogs show:
```
Jun 10 13:44:49 proxnode pvedaemon[2470]: <root@pam> successful auth for user 'prometheus@pve'
```
Permissions:
![image](https://user-images.githubusercontent.com/552230/41205690-34189602-6cb5-11e8-8832-511e0987f79f.png)

## Additional Notes

If I navigate to `/pve?target=192.168.1.2` on the exporter server I get:

```
Internal Server Error
'name'
```

**Comment by znerol** [comment]
Thanks for the report. I think that the [collector](https://github.com/znerol/prometheus-pve-exporter/blob/master/src/pve_exporter/collector.py) is trying to access a `name` field from a resource returned by the Proxmox API / Proxmoxer library. There are several places where a `name` field is expected. The reported error indicates that it is missing on one of the returned responses.

Do you run your the proxmox instances as a cluster?

**Comment by louisvernon** [comment]
Thanks for the quick response. Yes I am running the instances as a cluster. 

I dumped the returned `entry` values:
```
{u'name': u'WeAreDave', u'version': 13, u'quorate': 1, u'nodes': 3, u'type': u'cluster', u'id': u'cluster'}
{u'type': u'node', u'name': u'routerdave', u'level': u'', u'ip': u'192.168.1.2', u'nodeid': 4, u'online': 1, u'local': 0, u'id': u'node/routerdave'}
{u'name': u'superdave', u'level': u'', u'ip': u'192.168.1.3', u'nodeid': 1, u'local': 1, u'online': 1, u'type': u'node', u'id': u'node/superdave'}
{u'name': u'proxdave', u'level': u'', u'ip': u'192.168.1.4', u'nodeid': 2, u'local': 0, u'online': 1, u'type': u'node', u'id': u'node/proxdave'}
```
and `resource` values:
```
{u'node': u'proxdave', u'status': u'running', u'uptime': 64489, u'name': u'proxswarm', u'maxcpu': 4, u'diskread': 693779328, u'mem': 1286250496, u'vmid': 108, u'netin': 496933008, u'cpu': 0.0208175512826074, u'diskwrite': 1172274176, u'template': 0, u'netout': 371323692, u'disk': 0, u'type': u'qemu', u'id': u'qemu/108', u'maxdisk': 34359738368, u'maxmem': 4294967296}
{u'node': u'routerdave', u'status': u'running', u'uptime': 64455, u'name': u'routerswarm', u'maxcpu': 4, u'diskread': 693178240, u'mem': 1534328832, u'vmid': 107, u'netin': 442075169, u'id': u'qemu/107', u'diskwrite': 1458576384, u'template': 0, u'netout': 286157973, u'disk': 0, u'type': u'qemu', u'cpu': 0.0147186471673114, u'maxdisk': 34359738368, u'maxmem': 4294967296}
{u'node': u'superdave', u'status': u'running', u'uptime': 64459, u'name': u'superswarm', u'maxcpu': 4, u'diskread': 699204480, u'mem': 1519939584, u'vmid': 104, u'netin': 421317826, u'id': u'qemu/104', u'diskwrite': 1238469632, u'template': 0, u'netout': 326221937, u'disk': 0, u'type': u'qemu', u'cpu': 0.0119997486014583, u'maxdisk': 34359738368, u'maxmem': 4294967296}
...
```
Anything else I could do to help debug?

**Comment by znerol** [comment]
Honestly I have no idea. I might have the opportunity to test the exporter against a 5.2 cluster this week.

It is a bit unfortunate that the error message is so unspecific...

**Comment by znerol** [comment]
> It is a bit unfortunate that the error message is so unspecific...

I think this is due to the following fragment in [http.py](https://github.com/znerol/prometheus-pve-exporter/blob/982edc821f18842828ff84d7b66309f769522f19/src/pve_exporter/http.py#L97):

        try:
            return self._views[endpoint](**params)
        except Exception as error: # pylint: disable=broad-except
            self._errors.labels(args.get('module', 'default')).inc()
            raise InternalServerError(error)

I guess that the `try...except` block is hiding upstream exceptions. So maybe just remove the cruft so that the only line lift reads

        return self._views[endpoint](**params)

That might be something to try if you are comfortable hacking those files.

**Comment by louisvernon** [comment]
This seems to be related to the unusual state my cluster is in. Some time ago I removed a node (officedave) from the cluster without first deleting all linked VMs.
![image](https://user-images.githubusercontent.com/552230/41255668-850d46b4-6d84-11e8-82fe-ea692c7c857c.png)

https://github.com/znerol/prometheus-pve-exporter/blob/982edc821f18842828ff84d7b66309f769522f19/src/pve_exporter/collector.py#L237

At least some VMs still linked to this cluster were missing the 'name' attribute:
```
{u'node': u'officedave', u'vmid': 115, u'status': u'unknown', u'type': u'qemu', u'id': u'qemu/115'}
```
(In the above message I only grabbed a few lines of `resource` output)

I manually deleted all the configuration data related to this node within proxmox:
`rm -rf /etc/pve/nodes/officedave`

Now prometheus-pve-exporter is working great. Thanks for helping me debug this.
